cmd/diagnostic_test.go

```diff
@@ -171,26 +198,114 @@ token: mytoken
 				Eventually(session).Should(gexec.Exit(0))
 			})
 		})
+	})
 
-		Describe("whoami returns a user", func() {
-			BeforeEach(func() {
-				_, err := config.Write([]byte(`token: mytoken`))
-				Expect(err).ToNot(HaveOccurred())
-				Expect(config.Close()).To(Succeed())
-			})
+	Describe("whoami returns a user", func() {
+		var (
+			tempHome        string
+			command         *exec.Cmd
+			testServer      *ghttp.Server
+			defaultEndpoint = "graphql-unstable"
+			config          *os.File
+		)
 
-			It("print success", func() {
-				session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
-				Expect(err).ShouldNot(HaveOccurred())
-				Eventually(session.Err.Contents()).Should(BeEmpty())
-				Eventually(session.Out).Should(gbytes.Say(
-					fmt.Sprintf("API host: %s", defaultHost)))
-				Eventually(session.Out).Should(gbytes.Say(
-					fmt.Sprintf("API endpoint: %s", testServer.URL())))
-				Eventually(session.Out).Should(gbytes.Say("OK, got a token."))
-				Eventually(session.Out).Should(gbytes.Say("Hello, zzak."))
-				Eventually(session).Should(gexec.Exit(0))
-			})
+		BeforeEach(func() {
+			var err error
+			tempHome, err = ioutil.TempDir("", "circleci-cli-test-")
+			Expect(err).ToNot(HaveOccurred())
+
+			testServer = ghttp.NewServer()
+
+			const (
+				configDir  = ".circleci"
+				configFile = "cli.yml"
+			)
+
+			Expect(os.Mkdir(filepath.Join(tempHome, configDir), 0700)).To(Succeed())
+
+			config, err = os.OpenFile(
+				filepath.Join(tempHome, configDir, configFile),
+				os.O_RDWR|os.O_CREATE,
+				0600,
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = config.Write([]byte(`token: mytoken`))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(config.Close()).To(Succeed())
+
+			command = exec.Command(pathCLI,
+				"diagnostic",
+				"--skip-update-check",
+				"--host", testServer.URL())
+
+			command.Env = append(os.Environ(),
+				fmt.Sprintf("HOME=%s", tempHome),
+			)
+
+			query := `query IntrospectionQuery {
+		    __schema {
+		      queryType { name }
+		      mutationType { name }
+		      types {
+		        ...FullType
+		      }
+		    }
+		  }
+
+		  fragment FullType on __Type {
+		    kind
+		    name
+		    description
+		    fields(includeDeprecated: true) {
+		      name
+		    }
+		  }`
+
+			request := client.NewUnauthorizedRequest(query)
+			expected, err := request.Encode()
+			Expect(err).ShouldNot(HaveOccurred())
+
+			tmpBytes, err := ioutil.ReadFile(filepath.Join("testdata/diagnostic", "response.json"))
```

Comment:
Daniel pointed me to [golden.get](https://github.com/gotestyourself/gotest.tools/blob/b2aa35cc89847bde6179170bed441c491b3f0818/golden/golden.go#L25-L33) to read test files. It doesn't return an error and makes an assertion that the read was error-free.

Reply:
This was a result of test re-use, but I think we can replace these `ioutil.ReadFile()` calls in one fell swoop. (On a separate PR). :pray: